lib/checks/color/link-in-text-block-style-evaluate.js

```diff
@@ -26,18 +28,22 @@ export default function linkInTextBlockStyleEvaluate(node) {
 
   this.relatedNodes([parentBlock]);
 
-  if (elementIsDistinct(node, parentBlock)) {
-    return true;
-  }
-  if (hasPseudoContent(node)) {
-    this.data({ messageKey: 'pseudoContent' });
-    return undefined;
+  for (const inlineNode of inlineNodes) {
+    if (elementIsDistinct(inlineNode, parentBlock)) {
```

Comment:
I think checking each inline node individually is a little too simplified; it gives a false negative if an inline node specifies a sufficiently distinct style which is then overridden by an insufficiently distinct style in a descendant.

Here's an example of an integration test which ought to fail, but passes because of this issue:

```html
<p style="color: black">
  paragraph of text (fail: ancestor span is bolder than paragraph, but link reverts ancestor style)
  <span style="font-weight: bolder">
    <a
      style="text-decoration: none; color: #222; font-weight: normal"
      href="#"
      id="fail-ancestor-irrelevant-different-weight"
    >
      Link text</a
    >
  </span>
</p>
```

Probably this needs to do something different for cases where a child doesn't inherit/can't override the distinguishing property (like borders and text decoration) vs cases where a child can override the property (like font weight)